mlflow/tracking/_tracking_service/client.py

```diff
@@ -80,7 +81,34 @@ def get_metric_history(self, run_id, key):
 
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
         """
-        return self.store.get_metric_history(run_id=run_id, metric_key=key)
+
+        # NB: Paginated query support is currently only available for the RestStore backend.
+        # FileStore and SQLAlchemy store do not provide support for paginated queries and will
+        # raise an MlflowException if the `page_token` argument is not None when calling this
+        # API for a continuation query.
+
+        history = self.store.get_metric_history(
+            run_id=run_id,
+            metric_key=key,
+            max_results=GET_METRIC_HISTORY_MAX_RESULTS,
+            page_token=None,
+        )
+        if not isinstance(history, PagedList):
+            # Backwards compatible return if the server doesn't support paginated queries
+            return history
+        else:
+            metric_collection = history.to_list()
```

Comment:
```suggestion
        metric_collection = history.to_list()
```

nit

Reply:
```python
        if not isinstance(history, PagedList):
            ...
        else:
            metric_collection = history.to_list()
```

This looks like `metric_collection is undefined if isinstance(history, PagedList) is False` so I'd remove else.

Reply:
good point. short circuiting with the return.